File: synapse-transport/src/main/java/com/synapse/transport/feature/dataset/dto/queue/DatasetValidationJob.java
package com.synapse.transport.feature.dataset.dto.queue;

import java.util.UUID;

import com.synapse.transport.common.enums.DatasetProvider;
import com.synapse.transport.common.enums.ExperimentTaskType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DatasetValidationJob {
    private UUID workspaceId;
    private UUID experimentId;
    private ExperimentTaskType taskType;
    private DatasetProvider provider;
    private String datasetName;
    private String datasetConfigStorageKey;
}

File: synapse-runtime/common/models/queue/dataset_validation_job.py
from uuid import UUID
from pydantic import BaseModel, Field
from common.enums.dataset_provider import DatasetProvider
from common.enums.task_type import TaskType


class DatasetValidationJob(BaseModel):
    workspace_id: UUID = Field(alias="workspaceId")
    experiment_id: UUID = Field(alias="experimentId")
    task_type: TaskType = Field(alias="taskType")
    provider: DatasetProvider
    dataset_name: str = Field(alias="datasetName")
    dataset_config_storage_key: str = Field(alias="datasetConfigStorageKey")
